ops/pebble.py

```diff
@@ -386,11 +386,39 @@ def __repr__(self):
                 ).format(self=self)
 
 
+class Plan:
+    """Represents the effective Pebble configuration."""
+
+    def __init__(self, raw: str):
+        d = yaml.safe_load(raw) or {}
+        self._raw = raw
+        self._services = {name: Service(name, service)
+                          for name, service in d.get('services', {}).items()}
+
+    @property
+    def services(self):
+        """This plan's services mapping (maps service name to Service).
+
+        This property is currently read-only.
+        """
+        return self._services
+
+    def to_yaml(self) -> str:
+        """Return this plan's YAML representation.
+
+        This returns the exact YAML passed to Plan() without re-serialization,
+        that is, the YAML that Client.get_plan() received from Pebble.
```

Comment:
Why would we guarantee/document that?  Having it as an implementation detail and reserving ourselves the freedom to change that seems more useful.

Reply:
Fair point -- done.